diff --git a/Makefile b/Makefile
index eada865..27aaac0 100755
--- a/Makefile
+++ b/Makefile
@@ -51,8 +51,8 @@ CONFIG_STA_SUPPORT=y
 # Enable uAP mode support
 CONFIG_UAP_SUPPORT=y
 
-
-
+# Enable WIFIDIRECT support
+CONFIG_WIFI_DIRECT_SUPPORT=y
 
 # Re-association in driver
 CONFIG_REASSOCIATION=y
@@ -168,6 +168,9 @@ CONFIG_UAP_WEXT=n
 CONFIG_UAP_CFG80211=n
 endif
 
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
+	ccflags-y += -DWIFI_DIRECT_SUPPORT
+endif
 
 ifeq ($(CONFIG_MFG_CMD_SUPPORT),y)
 	ccflags-y += -DMFG_CMD_SUPPORT
@@ -510,10 +513,10 @@ ifeq ($(CONFIG_MLAN_WLS),y)
 mapp/mlanwls:
 	$(MAKE) -C $@
 endif
-#ifdef WIFI_DIRECT_SUPPORT
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
 mapp/wifidirectutl:
 	$(MAKE) -C $@
-#endif // WIFI_DIRECT_SUPPORT
+endif
 mapp/mlanevent:
 	$(MAKE) -C $@
 
@@ -541,18 +544,19 @@ ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/uaputl $@ INSTALLDIR=$(BINDIR)
 endif
 endif
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
+	cp -f README_WIFIDIRECT $(BINDIR)
+	cp -rpf script/wifidirect $(BINDIR)
+ifneq ($(APPDIR),)
+	$(MAKE) -C mapp/wifidirectutl $@ INSTALLDIR=$(BINDIR)
+endif
+endif
 ifeq ($(CONFIG_MLAN_WLS),y)
 	cp -f README_MLANWLS $(BINDIR)
 ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/mlanwls $@ INSTALLDIR=$(BINDIR)
 endif
 endif
-#ifdef WIFI_DIRECT_SUPPORT
-	cp -f README_WIFIDIRECT $(BINDIR)
-ifneq ($(APPDIR),)
-	$(MAKE) -C mapp/wifidirectutl $@ INSTALLDIR=$(BINDIR)
-endif
-#endif // WIFI_DIRECT_SUPPORT
 ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/mlanevent $@ INSTALLDIR=$(BINDIR)
 endif
@@ -590,12 +594,13 @@ ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/mlanwls $@ INSTALLDIR=$(BINDIR)
 endif
 endif
-#ifdef WIFI_DIRECT_SUPPORT
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
 	cp -f README_WIFIDIRECT $(BINDIR)
+	cp -rpf script/wifidirect $(BINDIR)
 ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/wifidirectutl $@ INSTALLDIR=$(BINDIR)
 endif
-#endif // WIFI_DIRECT_SUPPORT
+endif
 ifneq ($(APPDIR),)
 	$(MAKE) -C mapp/mlanevent $@ INSTALLDIR=$(BINDIR)
 endif
@@ -623,9 +628,9 @@ endif
 ifeq ($(CONFIG_MLAN_WLS),y)
 	$(MAKE) -C mapp/mlanwls $@
 endif
-#ifdef WIFI_DIRECT_SUPPORT
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
 	$(MAKE) -C mapp/wifidirectutl $@
-#endif // WIFI_DIRECT_SUPPORT
+endif
 	$(MAKE) -C mapp/mlanevent $@
 endif
 
@@ -665,9 +670,9 @@ endif
 ifeq ($(CONFIG_MLAN_WLS),y)
 	$(MAKE) -C mapp/mlanwls $@
 endif
-#ifdef WIFI_DIRECT_SUPPORT
+ifeq ($(CONFIG_WIFI_DIRECT_SUPPORT),y)
 	$(MAKE) -C mapp/wifidirectutl $@
-#endif // WIFI_DIRECT_SUPPORT
+endif
 	$(MAKE) -C mapp/mlanevent $@
 endif
 
