From e36e5902da07fc5828a105d7656e9bfeb3f710c6 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Mon, 11 Mar 2024 17:20:41 -0300
Subject: [PATCH] Patch for kernel 5.15

---
 bt/bt_sdiommc.c | 7 +++++++
 btxxx.mod       | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/bt/bt_sdiommc.c b/bt/bt_sdiommc.c
index 7f12c53..b9357c7 100755
--- a/bt/bt_sdiommc.c
+++ b/bt/bt_sdiommc.c
@@ -892,10 +892,17 @@ sd_download_firmware_w_helper(bt_private *priv)
 	PRINTM(MSG, "BT Request firmware: %s\n", cur_fw_name);
 	if (bt_extflg_isset(priv, EXT_BT_REQ_FW_NOWAIT)) {
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 32)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
+              ret = request_firmware_nowait(THIS_MODULE, FW_ACTION_UEVENT,
+                                            cur_fw_name, priv->hotplug_device,
+                                            GFP_KERNEL, priv,
+                                            sd_request_fw_callback);
+#else 
 		ret = request_firmware_nowait(THIS_MODULE, FW_ACTION_HOTPLUG,
 					      cur_fw_name, priv->hotplug_device,
 					      GFP_KERNEL, priv,
 					      sd_request_fw_callback);
+#endif
 #else
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 13)
 		ret = request_firmware_nowait(THIS_MODULE, FW_ACTION_HOTPLUG,
-- 
2.25.1

